src/wp-includes/class-wp-scripts.php

```diff
@@ -263,6 +272,31 @@ public function print_extra_script( $handle, $display = true ) {
 		return true;
 	}
 
+	/**
+	 * Processes the items and dependencies.
+	 *
+	 * Processes the items passed to it or the queue, and their dependencies.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string|string[]|false $handles Optional. Items to be processed: queue (false),
+	 *                                       single item (string), or multiple items (array of strings).
+	 *                                       Default false.
+	 * @param int|false             $group   Optional. Group level: level (int), no group (false).
+	 * @return string[] Array of handles of items that have been processed.
+	 */
+	public function do_items( $handles = false, $group = false ) {
+		$handles = false === $handles ? $this->queue : (array) $handles;
+		$this->all_deps( $handles );
+
+		// This statement is the only difference from parent::do_items().
+		if ( ! $this->printed_delayed_inline_script_loader && $this->has_delayed_inline_script( $this->to_do ) ) {
+			$this->print_delayed_inline_script_loader();
+		}
```

Comment:
This has to be moved here instead of being run externally at the `wp_print_scripts` action because we need to get access to an up-to-date `$this->to_do` which we only get after `$this->all_deps( $handles )` has been called. Otherwise, we can't rely on which delayed scripts were _actually_ enqueued and instead have to add the script if there are any delayed scripts registered at all.